### PR TITLE
Tech CORE-491 | improve UIDP - add logger and description to cli

### PR DESCRIPTION
**Pull request by nicholaspcr**
# Description

Kind: tech dep

Section: uidp/inprov

Summary: improve description of the cli commands in `inprov` and add loggers to the uidp handlers

## Changelog

> List the main points of change

## Pay attention to

> Important parts of the code that require special attention

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 
 # Changelog
 
+### #95 Tech CORE-491 | Improve UIDP
+- features:
+  - added a logger to the requests that the uidp handles.
+  - added description for the cli commands in the `inprov`.
+- fixes:
+  - removed the `COMMIT SUCESSED` message when the git hook didn't encounter any errors.
+
```
Comment (LucasStarlingdePaulaSalles) [inline CHANGELOG.md]: Missing --- at the end here

File: cmd/uid_provider/api/controller/routes.go [hunk 2]
```diff
@@ -4,6 +4,8 @@ import handler "inspr.dev/inspr/cmd/uid_provider/api/handlers"
 
 // initRoutes defines which routes the UID Provider API will have
 func (s *Server) initRoutes() {
+	logger.Debug("initializing Insprd server routes")
```
Comment (LucasStarlingdePaulaSalles) [inline cmd/uid_provider/api/controller/routes.go]: uidp server routes

File: cmd/uid_provider/api/handlers/handler.go [hunk 3]
```diff
@@ -38,12 +52,16 @@ func (h *Handler) CreateUserHandler() rest.Handler {
...
 		}
+
+		logger.Info("user created", zap.String("username", data.UID))
 	}).Post().JSON().Recover()
 }
 
 // DeleteUserHandler handles user deletion requests
 func (h *Handler) DeleteUserHandler() rest.Handler {
 	return rest.Handler(func(w http.ResponseWriter, r *http.Request) {
+		logger.Info("deleting a new user")
```
Comment (LucasStarlingdePaulaSalles) [inline cmd/uid_provider/api/handlers/handler.go]: deleting an user
deleting new user sounds counter intuitive

File: cmd/uid_provider/api/handlers/handler.go [hunk 4]
```diff
@@ -54,12 +72,16 @@ func (h *Handler) DeleteUserHandler() rest.Handler {
...
 		}
+
+		logger.Info("user deleted", zap.String("username", data.UID))
 	}).Delete().JSON().Recover()
 }
 
 // UpdatePasswordHandler handles requests to update an user password
 func (h *Handler) UpdatePasswordHandler() rest.Handler {
 	return rest.Handler(func(w http.ResponseWriter, r *http.Request) {
+		logger.Info("updating a new user")
```
Comment (LucasStarlingdePaulaSalles) [inline cmd/uid_provider/api/handlers/handler.go]: updating an user
same here

**Review by LucasStarlingdePaulaSalles — CHANGES_REQUESTED** [review]

**Review by LucasStarlingdePaulaSalles — APPROVED** [review]

**Review by luiz-couto — APPROVED** [review]

**Pull request merged**